ansible_base/authentication/authenticator_plugins/local.py

```diff
@@ -69,5 +87,84 @@ def authenticate(self, request, username=None, password=None, **kwargs):
+                return False
+            # Validate controller_user has a ldap_dn field, if it is not None, then the user is a local user
+            ldap_dn = controller_user.get("ldap_dn")
+            if (ldap_dn is None or ldap_dn != "") and user.password == "$encrypted$":
+                logger.warning(f"User '{username}' is unable to be confirmed as a local user.")
+                return False
+            return True
+        else:
+            return False
+
+    def _get_controller_user(self, username: str, password: str):
+        """
+        Get the user from the controller by making a request to the controller API /me/ endpoint.
+        If the user is not found, return None.
+        If the user is found, return the user.
+        """
+
+        controller_base_domain = get_setting('gateway_proxy_url')
+        if not controller_base_domain:
+            logger.warning("Controller authentication failed, unable to get controller base domain")
+            return None
+        controller_url = urljoin(controller_base_domain, "/api/controller/v2/me/")
+
+        try:
+            response = requests.get(controller_url, auth=HTTPBasicAuth(username, password), timeout=10)
+            response.raise_for_status()
+            user_data = response.json()
+
+            # Check if count exists and equals 1
+            count = user_data.get("count")
+            if count != 1:
+                logger.warning(f"Unable to authenticate user '{username}' with controller.")
+                return None
+
+            # Check if results exists and is a non-empty list
+            results = user_data.get("results")
+            if not results or not isinstance(results, list) or len(results) == 0:
+                logger.warning(f"Unable to authenticate user '{username}' with controller. Invalid or empty results.")
+                return None
+            return results[0]
+        except requests.exceptions.HTTPError as http_err:
+            logger.info(f"HTTP error occurred: {http_err}")
+            return None
+        except requests.exceptions.ConnectionError as conn_err:
+            logger.info(f"Connection error occurred: {conn_err}")
+            return None
+        except requests.exceptions.Timeout as timeout_err:
+            logger.info(f"Timeout error occurred: {timeout_err}")
+            return None
+        except requests.exceptions.RequestException as err:
+            logger.info(f"An unexpected error occurred: {err}")
+            return None
+        except ValueError as json_err:
+            logger.info(f"JSON decode error occurred: {json_err}")
+            return None
+
+    def update_gateway_user(self, username, password):
+        """
+        Update the gateway user with the controller credentials and set is_partially_migrated to False.
+        """
+        user = UserModel._default_manager.get_by_natural_key(username)
+        user.set_password(password)
+        user.save()
```

Comment:
```suggestion
        user.save(update_fields=['password', 'is_partially_migrated'])
```

What I imagine the sql of the two looks like:

save sql: `UPDATE "auth_user" SET "username" = 'newusername', "email" = 'existing_email', "password" = 'existing_password' WHERE "auth_user"."id" = 1;`

`update_fields` sql: `UPDATE "auth_user" SET  "password" = 'password-hash-here', "is_partially_migrated" = False WHERE "auth_user"."id" = 1;`

Reply:
Nice!! Did not know you could do this